File: UNIVERCITY_MANAGEMENT SYSTEM/UMI-API/publication/src/main/java/com/umi/publication/service/PublicationService.java
package com.umi.publication.service;
import com.umi.publication.dto.PublicationDTO;
import com.umi.publication.entity.Publication;

import java.util.List;


public interface PublicationService {
    List<PublicationDTO> getAllPublications();
    PublicationDTO getPublicationById(Long id);
    PublicationDTO getPublicationByAnnee(Long anneeId);
    PublicationDTO getPublicationByAuteur(Long auteurId);
    PublicationDTO getPublicationByEquipe(Long equipeId);
    PublicationDTO getPublicationByLabo(Long laboId);
    PublicationDTO addPublication(PublicationDTO publicationDTO);
    PublicationDTO updatePublication(Long id, PublicationDTO publicationDTO);
    void deletePublication(Long id);}

File: UMI-CLIENT/src/app/services/publication.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Publication } from './../models/publication.model';

@Injectable({
  providedIn: 'root'
})
export class PublicationService {
 
  private apiUrl = 'http://localhost:9015/api/publications';

  constructor(private http: HttpClient) { }

  getAllPublications(): Observable<Publication[]> {
    return this.http.get<Publication[]>(this.apiUrl);
  }
  getPublicationById(id: number): Observable<Publication> {
    return this.http.get<Publication>(`${this.apiUrl}/${id}`);
  }

  getPublicationByAnnee(id: number): Observable<Publication> {
    return this.http.get<Publication>(`${this.apiUrl}/annee/${id}`);
  }
  getPublicationByEquipe(id: number): Observable<Publication> {
    return this.http.get<Publication>(`${this.apiUrl}/equipe/${id}`);
  }
  getPublicationByAuteur(id: number): Observable<Publication> {
    return this.http.get<Publication>(`${this.apiUrl}/auteur/${id}`);
  }
  getPublicationByLabo(id: number): Observable<Publication> {
    return this.http.get<Publication>(`${this.apiUrl}/labo/${id}`);
  }


  addPublication(publication: Publication): Observable<Publication> {
    return this.http.post<Publication>(this.apiUrl, publication);
  }

  updatePublication(id: number, publication: Publication): Observable<Publication> {
    return this.http.put<Publication>(`${this.apiUrl}/${id}`, publication);
  }

  deletePublication(id: number): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/${id}`);
  }

}
